jvm-packages/xgboost4j-spark/src/test/scala/ml/dmlc/xgboost4j/scala/spark/SparkParallelismTrackerSuite.scala

```diff
@@ -0,0 +1,52 @@
+/*
+ Copyright (c) 2014 by Contributors
+
+ Licensed under the Apache License, Version 2.0 (the "License");
+ you may not use this file except in compliance with the License.
+ You may obtain a copy of the License at
+
+ http://www.apache.org/licenses/LICENSE-2.0
+
+ Unless required by applicable law or agreed to in writing, software
+ distributed under the License is distributed on an "AS IS" BASIS,
+ WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ See the License for the specific language governing permissions and
+ limitations under the License.
+ */
+
+package ml.dmlc.xgboost4j.scala.spark
+
+import ml.dmlc.xgboost4j.java.XGBoostError
+import org.apache.spark.rdd.RDD
+import org.scalatest.FunSuite
+
+class SparkParallelismTrackerSuite extends FunSuite with PerTest {
```

Comment:
Consider not inheriting `PerTest` if its semantics are not applicable.

Reply:
The problem is not the inheritance. There is `sparkSession` is shared among all tests unless it's explicitly stopped.

Reply:
You're right. My point point was that you could manually create a session within each test instead of inheriting `PerTest`.

Reply:
Done.